operations/mimir-mixin-compiled/alerts.yaml

```diff
@@ -1026,6 +1026,18 @@ groups:
           for: 5m
           labels:
             severity: critical
+        - alert: MimirKafkaClientBufferedProduceBytesTooHigh
+          annotations:
+            message: Mimir {{ $labels.pod }} in {{ $labels.cluster }}/{{ $labels.namespace }} Kafka client produce buffer utilization is {{ printf "%.2f" $value }}%.
+            runbook_url: https://grafana.com/docs/mimir/latest/operators-guide/mimir-runbooks/#mimirkafkaclientbufferedproducebytestoohigh
+          expr: |
+            max by(cluster, namespace, pod) (max_over_time(cortex_ingest_storage_writer_buffered_produce_bytes{quantile="1.0"}[1m]))
+            /
+            min by(cluster, namespace, pod) (min_over_time(cortex_ingest_storage_writer_buffered_produce_bytes_limit[1m]))
+            * 100 > 50
+          for: 5m
+          labels:
+            severity: warning
```

Comment:
Why a warning? This should be problematic enough to warrant a decent level of criticality. Perhaps we should bump the threshold and set it to critical? I'll leave the decision in your hands.

Reply:
I was also in doubt whether it should be a warning or critical. Once the buffer is full, SLO alerts should fire, but if we address this issue when the buffer is < 100% then we may avoid a bigger issue. I changed the alert severity to critical. We can always revisit the alert if, for any reason I can't foresee, this will turn out to be noisy or false positive.